### PR TITLE
Update Javadoc and visibility of classes

### PR DESCRIPTION
**Pull request by Michael1993**
Proposed commit message:

```
Update Javadoc #576 / #580

Update the slightly out-of-date or missing Javadoc.

Closes: #576
PR: #580
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [X] There is documentation (Javadoc and site documentation; added or updated)

Contributing
* [X] A prepared commit message exists

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

File: src/main/java/org/junitpioneer/vintage/Test.java [hunk 1]
```diff
@@ -44,6 +44,9 @@
 @org.junit.jupiter.api.Test
 public @interface Test {
 
+	/**
+	 * Dummy default class for the `expected` parameter.
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/vintage/Test.java:48]: Markdown doesn't work here. 😉

```suggestion
	 * Dummy default class for the <code>expected</code> parameter.
```

File: src/main/java/org/junitpioneer/jupiter/CartesianProductTest.java [hunk 2]
```diff
@@ -97,7 +97,9 @@
 	 * Class for defining sets to a {@code CartesianProductTest} execution.
 	 *
 	 * @since 1.0
+	 * @deprecated CartesianProductTest has been superseded by CartesianTest
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/CartesianProductTest.java:100]: Just as in line 52:

```suggestion
	 * @deprecated CartesianProductTest has been superseded by CartesianTest, scheduled to be removed in 2.0
```

File: src/main/java/org/junitpioneer/jupiter/CartesianEnumSource.java [hunk 3]
```diff
@@ -183,9 +183,15 @@ private static void validatePatterns(CartesianEnumSource enumSource, Set<? exten
 
 	}
 
+	/**
+	 * Containing annotation of repeatable {@code CartesianEnumSource}
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/CartesianEnumSource.java:187]: Below we end the sentence with a period (.):

```suggestion
	 * Containing annotation of repeatable {@code CartesianEnumSource}.
```

File: src/main/java/org/junitpioneer/jupiter/CartesianValueSource.java [hunk 4]
```diff
@@ -99,9 +99,15 @@
 	 */
 	Class<?>[] classes() default {};
 
+	/**
+	 * Containing annotation of repeatable {@code CartesianValueSource}
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/CartesianValueSource.java:103]: Below we end the sentence with a period (.):

```suggestion
	 * Containing annotation of repeatable {@code CartesianValueSource}.
```

File: src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java [hunk 5]
```diff
@@ -31,7 +31,7 @@
 import org.opentest4j.AssertionFailedError;
 import org.opentest4j.TestAbortedException;
 
-public class RetryingTestExtension implements TestTemplateInvocationContextProvider, TestExecutionExceptionHandler {
+class RetryingTestExtension implements TestTemplateInvocationContextProvider, TestExecutionExceptionHandler {
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/RetryingTestExtension.java:34]: Good catch (also for `DisableIfNameExtension`)! 👍

**Review by beatngu13 — CHANGES_REQUESTED** [review]

**Review by beatngu13 — APPROVED** [review]

**Review by Bukama — APPROVED** [review]
Thank you!

**Pull request merged**